Removed use of NodeJS. Utilized LGWebOSRemote for WebOS 2.0+

akhan23wgu opened the pull request:
Removed NodeJS. Using LGWebOSRemote CLI allows for faster command execution - almost instantly.

akhan23wgu committed:
Using LGWebOSRemote CLI in place of WoL and NodeJS

See klattimer/LGWebOSRemote for Remote CLI interface

akhan23wgu committed:
Wrote wrong file, this is the correct one

akhan23wgu committed:
Update README.md

akhan23wgu committed:
Removed need for hardcoded mac address

akhan23wgu committed:
Update README.md

akhan23wgu committed:
Update README.md

akhan23wgu committed:
Added support for applications and volume controls

Alexa turn on/off "tv", "plex", "netflix", and "volume" commands are now supported.

akhan23wgu committed:
Update README.md

akhan23wgu committed:
Update README.md

akhan23wgu committed:
Update alexa-tv.py

Added inputMediaPlay and inputMediaPause functions to resume and pause what's playing, call-sign is "playback". Playback can also be used as an "OK" button in Netflix stuck on an "Are you still watching?" prompt.

akhan23wgu committed:
Update README.md

Added Playback functionality.

"Alexa, turn on Playback"
"Alexa, turn off Playback"

efpe commented:
I love this, will take a look if I have a minute. Thanks for the PR!

karlinnolabs commented:
You're importing the lgtv.py from my project but I don't think you want to do that - if I make changes to LGWebOSRemote then you'll need to pull in another copy - or miss out on the changes, I'd suggest using a git subproject instead - or pip but I'd need to sort out lgtv to work with pip for that.

efpe commented:
@karlinnolabs I would go with pip, that would be much cleaner that using submodules.

karlinnolabs commented:
OK, I'll see about setting up the repo with pypi tonight

TRingo1 commented:
When were you going to merge to master? just wondering, if all tested and working yet.

karlinnolabs commented:
I don't think it should merge until I've done the pypi work, and this patch is reworked to reflect that.

akhan23wgu committed:
Update README.md

Added additional detail on installation

akhan23wgu committed:
Update alexa-tv.py

Fixed default volume from 4 to 34

akhan23wgu committed:
Update alexa-tv.py

Replaced 'mute' with 'volume 0'.

Set default volume level to 44.

akhan23wgu committed:
Update README.md

Added custom commands and modified items to reflect updated changes in alexa-tv.py.

klattimer commented:
LGTV is now usable as a library. I need to make a pypi release of it when I find a minute but it's pip installable from the git URL now.

pip install git+https://github.com/klattimer/LGWebOSRemote.git

klattimer commented:
@akhan23wgu can you update the patch.

ganeyvim commented:
python lgtv.py setInput 
What is the input ID for Live TV?
Nothing I tried, works. All the others are fine (HDMI_1, netflix, av1, etc)
I cannot for the life of me get in back to Antenna TV, keep getting an error of
{"payload": {"errorCode": -1000, "errorText": "expected inputId", "returnValue": false}, "type": "error", "id": "0", "error": "500 Application error"}

akhan23wgu commented:
@klattimer What would be the best way to pull from the library? After pip install do I just simply do a "from LGTV import LGTVScan, LGTVClient, getCommands" How would I call commands, vs calling the script as I am doing right now, or do I continue to call the script?

akhan23wgu committed:
Update README.md

Updated requirements

karlinnolabs commented:
@ganeyvim you're asking on the wrong project, please go to the actual project to report an issue.

karlinnolabs commented:
@akhan23wgu take a look at the command line utility - basically that's what it's doing. https://github.com/klattimer/LGWebOSRemote/blob/master/lgtv.py

akhan23wgu committed:
Update README.md

Only works with the Echo Dot. ¯\_(ツ)_/¯
